os/arch/arm/src/armv7-m/up_assert.c

```diff
@@ -440,11 +443,27 @@ static void _up_assert(int errorcode)
 		request_msg.cmd = BINMGR_FAULT;
 		request_msg.pid = getpid();
 
+		if (current_regs) {
+			tcb = sched_self();
+			sched_removereadytorun(tcb);
+#if CONFIG_RR_INTERVAL > 0
+			tcb->timeslice = 0;
+#endif
+			tcb->sched_priority = SCHED_PRIORITY_MIN;
+			sched_addreadytorun(tcb);
+		}
+
 		/* Send a message to fault manager with pid of the faulty task */
 		ret = mq_send(binmgr_mq, (const char *)&request_msg, sizeof(binmgr_request_t), BINMGR_FAULT_PRIO);
 		DEBUGASSERT(ret == 0);
+
+		if (current_regs) {
+			tcb = sched_self();
+			current_regs = tcb->xcp.regs;
+		}
 		mq_close(binmgr_mq);
-		exit(errorcode);
```

Comment:
This is not removed from armv7-r/arm_assert.c. Could you remove it with different commit?

Reply:
yes, as the behaviour of abort/hard fault are different in R and M, we will verify and submit for R in separate PR.